src/internal/operations/backup.go

```diff
@@ -632,23 +633,98 @@ func getNewPathRefs(
 	return newPath, newLoc, updated, nil
 }
 
-func mergeDetails(
+// mergeDetailsFromAssistBackups merges the deets from the assist backups into
+// the current deets.
+func mergeDetailsFromAssistBackups(
 	ctx context.Context,
 	detailsStore streamstore.Streamer,
-	backups []kopia.BackupEntry,
-	dataFromBackup kopia.DetailsMergeInfoer,
 	deets *details.Builder,
-	writeStats *kopia.BackupStats,
+	assistBackups []kopia.BackupEntry,
 	serviceType path.ServiceType,
 	errs *fault.Bus,
-) error {
-	detailsModel := deets.Details().DetailsModel
+) (map[string]details.Entry, error) {
+	// If the backup operation did not produce any new details, or if we don't
+	// have any assist backups, we don't need to merge anything.
+	if deets.Empty() || len(assistBackups) == 0 {
+		logger.Ctx(ctx).Infow("no assist backups to merge",
+			"deets_empty", deets.Empty(),
+			"num_assist_backups", len(assistBackups))
+
+		return nil, nil
+	}
 
-	// getting the values in writeStats before anything else so that we don't get a return from
-	// conditions like no backup data.
-	writeStats.TotalNonMetaFileCount = len(detailsModel.FilterMetaFiles().Items())
-	writeStats.TotalNonMetaUploadedBytes = detailsModel.SumNonMetaFileSizes()
+	repoRefToAssistDeets := make(map[string]details.Entry)
+
+	for _, assistBackup := range assistBackups {
+		mctx := clues.Add(ctx, "assist_backup_id", assistBackup.ID)
 
+		baseDeets, err := getDetailsFromBackup(
+			mctx,
+			assistBackup.Backup,
+			detailsStore,
+			errs)
+		if err != nil {
+			return nil, clues.New("fetching details for assist backup")
+		}
+
+		for _, entry := range baseDeets.Items() {
+			rr, err := path.FromDataLayerPath(entry.RepoRef, true)
+			if err != nil {
+				return nil, clues.New("parsing base item info path").
+					WithClues(mctx).
+					With("repo_ref", path.NewElements(entry.RepoRef))
+			}
+
+			// Although this assist base has an entry it may not be the most recent.
+			// Check the reasons a snapshot was returned to ensure we only choose the
+			// recent entries.
+			if !matchesReason(assistBackup.Reasons, rr) {
```

Comment:
@ashmrtn I want to address one case here that you had brought up.

> assist bases `F1` (oldest, original categories: [email, contacts], selected category: `[contacts]`) & `F2` (original category: [email], selected category: `[email]`), have different versions of the item.

I think we are okay here. Repo ref for an email item from `F1` will not match with the selected `F1` reason(`assistBackup.Reasons = contacts`) here. So this item will be discarded. 

This email item will be picked up from `F2`.

Reply:
ooo good call. I want to take some time to think about this more in the context of concurrent backups, but that logic definitely makes sense if none of the assist bases overlapped

Reply:
Thought about concurrent backups with overlapping reasons a bit more. Even if we address that, it doesn't give us a foolproof solution(see scenario below). I think we should proceed with this PR with a followup to potentially support concurrent backup cases. 
I'll write up an internal doc to discuss this topic. 

**Scenario:**

* Extending on the example in this comment thread. Say `C3[contacts]` follows `F1` & `F2` and finishes.
* Now lets say `F4[email, contact]` follows `C3`. During base selection, `F2` will be selected for `email` as an assist base. But there will be no assist base for `contacts` because `C3[contacts]` is most recent. 
* This means the `email` item which was more recent in `F1`(compared to `F2`) will not be available for merge here.

Reply:
Sounds good! 

> Part of what makes it (a tad) easier to deal with the existing code is that chunks of it are agnostic to assumptions like that.

100% with you on that.